xdsl/dialects/experimental/air.py

```diff
@@ -154,7 +154,7 @@ class ChannelPutOp(IRDLOperation):
     src_sizes = var_operand_def(IndexType())
     src_strides = var_operand_def(IndexType())
 
-    async_token = opt_result_def(AsyncTokenAttr())
+    async_token = result_def(AsyncTokenAttr())
```

Comment:
Should these be in this PR?

Reply:
These tests fail with my changes. This is because we silently always set variadic (meaning also optional) results to have length 1. Our filecheck passes whether or not the results are present in the CHECK lines, because we match on substrings, so we silently ignore whether there are 0 or 1 results. Without the change on this line, the tests crash with the `NotImplementedError` I added in this PR.